staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go

```diff
@@ -146,13 +121,27 @@ func Run(g *Generator) {
 
 	boilerplate, err := g.Common.LoadGoBoilerplate()
 	if err != nil {
-		log.Fatalf("Failed loading boilerplate: %v", err)
+		log.Fatalf("Failed loading boilerplate (consider using the go-header-file flag): %v", err)
 	}
 
 	protobufNames := NewProtobufNamer()
 	outputPackages := generator.Packages{}
 	nonOutputPackages := map[string]struct{}{}
-	for _, d := range strings.Split(g.Packages, ",") {
+
+	apimachineryPackages := strings.Split(g.APIMachineryPackages, ",")
+	if len(apimachineryPackages) == 0 && len(apimachineryPackages[0]) == 0 {
```

Comment:
This if is strange - if len(apimachineryPackage) == 0, then we shouldn't check apimachineryPackages[0] - it will crash.

Reply:
must be ||

Reply:
let me compare before split. it's simpler.

Reply:
fixed